### PR TITLE
Delay import of pkg_resources

### PR DESCRIPTION
**Pull request by cgohlke**
Importing pkg_resources is extremely slow.

See also #758

It would be better not to use pkg_resources at all.

# Description

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Delaying the import is fine with me, apologies for the regression, and thanks for the PR fix. Curious @tlambert03 if you can see any alternatives?

**Comment by tlambert03** [comment]
thanks @cgohlke, and sorry about that.  @sofroniewn, yes, lemme just reimplement that function and not use pkg_resources.  I'll push to this PR

**Review by tlambert03 — APPROVED** [review]

File: napari/_qt/dialogs/qt_plugin_dialog.py [hunk 1]
```diff
@@ -40,6 +39,18 @@
...
+    import packaging.version
+
+
+def parse_version(v) -> 'packaging.version._BaseVersion':
+    import packaging.version
+
+    try:
+        return packaging.version.Version(v)
+    except packaging.version.InvalidVersion:
+        return packaging.version.LegacyVersion(v)
```
Comment (sofroniewn) [inline napari/_qt/dialogs/qt_plugin_dialog.py]: Not sure if this is still a WIP, but should we define this in a util file somewhere and import it here. It seems fairly reusable?

Comment (tlambert03) [reply napari/_qt/dialogs/qt_plugin_dialog.py]: can if you want to... got an idea of where?

Comment (sofroniewn) [reply napari/_qt/dialogs/qt_plugin_dialog.py]: i don't have strong feelings. `napari/utils/misc` works fine for me for now. can always then reorganize.

**Comment by sofroniewn** [comment]
Looks like the windows tests fails are independent of this PR, so I will merge now. Thanks all

**Comment by Czaki** [comment]
What did you think about adding some tests for future? some subprocess call with a file like: 

```
import sys

import napari 

assert "pkg_resources" not in sys.modules
```
?

**Comment by sofroniewn** [comment]
> What did you think about adding some tests for future? some subprocess call with a file like:

i'm open to a test like that

**Pull request merged**